ui/src/dashboards/components/CellEditorOverlay.tsx

```diff
@@ -409,47 +408,38 @@ class CellEditorOverlay extends Component<Props, State> {
   }
 
   private getConfig = async (
-    url,
+    source: SourcesModels.Source,
     id: string,
     query: string,
     templates: Template[]
   ): Promise<QueriesModels.QueryConfig> => {
-    // replace all templates but :interval:
-    query = replaceTemplate(query, templates)
-    let queries = []
-    let durationMs = DEFAULT_DURATION_MS
-
     try {
-      // get durationMs to calculate interval
-      queries = await getQueryConfigAndStatus(url, [{query, id}])
-      durationMs = _.get(queries, '0.durationMs', DEFAULT_DURATION_MS)
-
-      // calc and replace :interval:
-      query = replaceInterval(query, DEFAULT_PIXELS, durationMs)
+      query = await replaceTempVars(query, source, templates, DEFAULT_PIXELS)
     } catch (error) {
       console.error(error)
       throw error
     }
 
     try {
       // fetch queryConfig for with all template variables replaced
-      queries = await getQueryConfigAndStatus(url, [{query, id}])
+      const queries = await getQueryConfigAndStatus(source.links.queries, [
```

Comment:
looks cleaner -- so we don't need to make two requests and replace in between?

Reply:
@jaredscheib  The first request on `417` is used to extract and then replace the query `:interval:` with a duration. Previously, `getQueryConfigAndStatus` was being called and then implementing an interval replacement scheme similar to the `replace` function in `query.ts`.

Reply:
👍 as long as it achieves the same effect, looks good to me